library/src/jvmTest/kotlin/nl/bijdorpstudio/kiban/IbanFieldsTest.kt

```diff
@@ -31,12 +30,19 @@ class IbanFieldsTest {
             .forAll { td ->
                 val iban = Iban.parse(td.plain)
                 val branchIdentifier = IBANFields.getBranchIdentifier(iban)
-                assertThat(branchIdentifier).isNotNull()
                 if (td.branch == null) {
                     assertThat(branchIdentifier.isPresent).isFalse()
                 } else {
                     assertThat(branchIdentifier.get()).isEqualTo(td.branch)
                 }
             }
     }
+
+    // This should fail lint
+    @Test
+    fun test1() {
+       val test: Boolean? = true
+
+        assertThat(test!!).isTrue()
```

Comment:
Only boolean binary expressions will fail right now. It's totally reasonable to use `isTrue` or `isFalse` on boolean properties, but usually there's a better assertion if you're using a boolean expression.

If you change this line to the following, it should fail.

```
assertThat(test != null).isTrue()
```

Reply:
I just did a quick sanity check on that, and am not seeing the failure with the tweak, either. It might be a problem with the lint plugin. I'll dig a little more.